### ISSUE TITLE
Добавить NRF24L01 с усилителем

### ISSUE THREAD

**Issue by mforis** [halny123/sportiduino#2]
Добрый день! Есть идея добавить NRF24L01, чтобы она могла передавать сплит, который есть в чипе и производила отметку, то есть, участник проходит дистанцию отмечается на базе отметки и бежит на следующий КП, данные с его чипа передаются на финиш и можно вывести на экран данные. И подвести примерные результаты. Нужна помощь по созданию прошивки и софта. Спасибо!

**Comment by UrikEEE** [comment]
С интересом слежу за проектом sportiduino, в том числе и за этой веткой. **mforis**, хоть ваш вопрос адресован не мне, но я не совсем понял, что же вы хотите? Куда хотели бы добавить чип NRF24L01? В схему базовой станции (но он там и так есть, только запитывается через замкнутые контакты геркона) или же сделать чип интеллектуальным (чтобы это была не просто rfid-метка, а содержала еще свой микроконтроллер, NRF24L01 и питание)?

**Comment by mforis** [comment]
@UrikEEE, нет, я не о чипе. А например, проходят соревнования, человек проходит дистанцию, ну и допустим он берет 10 КП из 20 и бежит дальше, но помимо отметки в чип, база считывает информацию о пройденных КП и передаёт эту информацию на финиш с помощью NRF24L01. На финише отображается временный сплит и временное место.

**Comment by UrikEEE** [comment]
**mforis**, спасибо за разъяснения, теперь понятен ваш ваш вопрос. Выскажу свое мнение. Если кратко: в таком виде (считать все отметки из чипа и передать их через NRF24L01 на финиш) такой объем информации передать маловероятно, да и не за чем. Теперь подробнее. Просто прикиньте объем информации для передачи. Одна отметка - это минимум 7 байт (номер КП - 1 байт, номер чипа - 2 байта, дата/время отметки - 4 байта), количество отметок в чипе - величина переменная (может быть несколько десятков, а то и больше). Идем дальше: сама отметка сильно затянется во времени (надо считать в память AVR все отметки из чипа), будет ли столько свободной памяти - тоже вопрос. Затем надо эти отметки передать. NRF24L01 обменивается пакетами длиной не более 32 байт, надо будет "рожать" протокол обмена. Про дальность обмена речи даже не веду. С усилителем в чистом поле до 1 километра заявляют, но как оно будет в реальности в лесу (а если на пересеченке)? И самое главное - зачем на финише нужны все отметки из чипа (при отметке на промежуточном КП) ? Разве недостаточно будет данных об отметке только на самом промежуточном КП? Базовая станция при отметке записывает в чип номер станции и время отметки, если к ним добавить еще номер станции и одним пакетом пульнуть в NRF24L01 - это я понимаю. Фиксированный размер пакета, простота реализации, минимум затрат времени и памяти. Ну и дополнительный расход энергии батареек тоже минимальный.

Подумал и вспомнил, где может пригодиться передача всех отметок из чипа! В ориентировании (в российском точно) есть такая дисциплина, как маркировка. И по новым правилам на ней может быть электронная отметка и может быть также эстафета с маркировкой. Так вот, участник в конце своего этапа должен пробежать оценочный круг, за время прохождения которого судьи должны подсчитать количество штрафных кругов. Как именно это делать с электронной отметкой - пока не регламентировано. Вот если на пункте оценки поставить базовую станцию, которая будет считывать все отметки из чипа (фактически это считывающая станция, ей даже отметку в чип можно не делать) и передавать их на финиш - то тогда эта идея подойдет. Но опять же - слишком высока цена ошибки. Случись какой сбой в передаче данных или там другие какие-то проблемы - то времени на их устранение катастрофически мало (время прохождения оценочного круга), а какой скандал будет, если к концу оценочного круга не будет подсчитан штраф! Если только в качестве эксперимента...

**Comment by fimich1** [comment]
экспериментировали, примерно такой вариант но с модулем LoRa Wan, в качестве альтернативы радио-кп - хороший вариант, пока отложили до завершения других проектов..

**Comment by mforis** [comment]
@fimich1 , понимаю, мы с коллегой планируем сделать бесконтактную отметку. В станцию отметки добавить NRF24L01 (без усиления) и разработать бесконтактный чип на микроконтроллере NRF52832 (по моим расчетам, стоимость чипа, примерно 5$)

**Comment by fimich1** [comment]
есть мнение, что бесконтактная отметка имеет смысл только если есть возможность отсекать финиш лыжного ориентирования в воротах 9 метров и вело, по типу петли в sportident

**Comment by mforis** [comment]
@fimich1 , мы что-то подобное и планируем для ТПМ
